fog/ingest/server/src/controller.rs

```diff
@@ -1028,11 +1071,34 @@ where
         ingress_public_key: &CompressedRistrettoPublic,
         state: &mut MutexGuard<IngestControllerState>,
     ) -> Result<IngressPublicKeyStatus, Error> {
-        self.update_enclave_report_cache()?;
```

Comment:
removing this line, or rather, making it conditional on the current report not matching the key we are supposed to be publishing, is the main point of this PR